src/components/component.ts

```diff
@@ -356,15 +360,18 @@ export class Component {
 
   private _generateClipPath() {
     // The clip path will prevent content from overflowing its Component space.
-    // HACKHACK: IE <=9 does not respect the HTML base element in SVG.
+    this._clipPathID = Utils.DOM.generateUniqueClipPathId();
+    let clipPathParent = this._boxContainer.append("clipPath").attr("id", this._clipPathID);
+    this._addBox("clip-rect", clipPathParent);
+    this._updateClipPath();
+  }
+
+  private _updateClipPath() {
```

Comment:
That's the only place where things are drawn. Subclasses should be calling `_renderImmediately()`, and then calling their own logic inside that -- I've filed https://github.com/palantir/plottable/issues/2771 for classes that aren't doing this properly. I fixed it for `Plot` and `DragBoxLayer`; the other two (`Axes` classes) aren't using `clipPath`.